beacon-chain/slasher/chunks.go

```diff
@@ -0,0 +1,553 @@
+//
+// Recall that for an epoch, e, min((att.target - e) for att in attestations where att.source > e)
+// That is, it is the minimum distance between the specified epoch and all attestation
+// target epochs a validator has created where att.source.epoch > e.
+//
+// Recall that a MinSpanChunksSlice struct represents a single slice for a chunk index
+// from the collection below:
+//
+//                                     val0    val1    val2
+//                                     {  }    {  }    {  }
+//   chunk_0_for_validators_0_to_2 = [[2, 2], [2, 2], [2, 2]]
+//
+//                                     val0    val1    val2
+//                                     {  }    {  }    {  }
+//   chunk_1_for_validators_0_to_2 = [[2, 2], [2, 2], [2, 2]]
+//
+//                            ...
+//
+//                                     val0    val1    val2
+//                                     {  }    {  }    {  }
+//   chunk_N_for_validators_0_to_2 = [[2, 2], [2, 2], [2, 2]]
+//
+// Let's take a look at how this update will look for a real set of min span chunk:
+// For the purposes of a simple example, let's set H = 2, meaning a min span
+// will hold 2 epochs worth of attesting history. Then we set C = 2 meaning we will
+// chunk the min span into arrays each of length 2.
+//
+// So assume we get an epoch 4 and validator 0, then, we need to update every epoch in the span from
+// 4 down to 2. First, we find out which chunk epoch 4 falls into, which is calculated as:
+// chunk_idx = (epoch % H) / C = (4 % 2) / 2 = 0
+//
+//
+//                                     val0    val1    val2
+//                                     {  }    {  }    {  }
+//   chunk_0_for_validators_0_to_3 = [[2, 2], [2, 2], [2, 2]]
+//                                     |
+//                                     |-> epoch 4 for validator 0
+//
+// Next up, we proceed with the update process for validator index 0, starting epoch 4
+// all the way down to epoch 2. We will need to go down the array as far as we can get. If the
+// lowest epoch we need to update is < the lowest epoch of a chunk, we need to proceed to
+// a different chunk index.
+//
+// Once we finish updating a chunk, we need to move on to the next chunk. This function
+// returns a boolean named keepGoing which allows the caller to determine if we should
+// continue and update another chunk index. We stop whenever we reach the min epoch we need
+// to update, in our example, we stop at 2, which is still part of chunk 0, so no need
+// to jump to another min span chunks slice to perform updates.
+func (m *MinSpanChunksSlice) Update(
+	args *chunkUpdateArgs,
+	validatorIndex types.ValidatorIndex,
+	startEpoch,
+	newTargetEpoch types.Epoch,
+) (keepGoing bool, err error) {
+	// The lowest epoch we need to update. This is a sliding window from (current epoch - H) where
+	// H is the history length a min span for a validator stores.
+	var minEpoch types.Epoch
```

Comment:
Explicitly setting the value to `0` here would make the code more readable in my opinion.